core/block_processor.go

```diff
@@ -81,8 +81,10 @@ func (sm *BlockProcessor) TransitionState(statedb *state.StateDB, parent, block
 	return receipts, nil
 }
 
-func (self *BlockProcessor) ApplyTransaction(coinbase *state.StateObject, statedb *state.StateDB, header *types.Header, tx *types.Transaction, usedGas *big.Int, transientProcess bool) (*types.Receipt, *big.Int, error) {
-	cb := statedb.GetStateObject(coinbase.Address())
+func (self *BlockProcessor) ApplyTransaction(statedb *state.StateDB, header *types.Header, tx *types.Transaction, usedGas *big.Int, transientProcess bool) (*types.Receipt, *big.Int, error) {
+	cb := statedb.GetOrNewStateObject(header.Coinbase)
+	cb.SetGasLimit(header.GasLimit)
```

Comment:
Yepp we got to make sure gasUsed (gasPool) is correctly tracked, closing this PR and let's re-open a new to develop branch once we figure out the correct tracking.